src/main/java/teammates/common/datatransfer/StudentAttributes.java

```diff
@@ -149,12 +148,13 @@ public String getPublicProfilePictureUrl() {
     }
 
     public boolean isEnrollInfoSameAs(StudentAttributes otherStudent) {
-        return (otherStudent != null) && otherStudent.email.equals(this.email)
+        return (otherStudent != null) 
+                && otherStudent.email.equals(this.email)
                 && otherStudent.course.equals(this.course)
-                && otherStudent.name.equals(this.name)
-                && otherStudent.comments.equals(this.comments)
-                && otherStudent.team.equals(this.team)
-                && otherStudent.section.equals(this.section);
+                && StringHelper.recoverFromSanitizedText(otherStudent.name).equals(this.name)
+                && StringHelper.recoverFromSanitizedText(otherStudent.comments).equals(this.comments)
+                && StringHelper.recoverFromSanitizedText(otherStudent.team).equals(this.team)
+                && StringHelper.recoverFromSanitizedText(otherStudent.section).equals(this.section);
```

Comment:
Instead of doing this, it might be better to tackle the underlying problem, #2801.